src/components/counter/counter.js

```diff
@@ -0,0 +1,31 @@
+import React from "react";
+import "./counter.css";
+
+class Counter extends React.Component{
+    state = {
+        counterValue : 0
+    };
+
+incrementCounter(){
+    let initialValue = this.state.counterValue;
+    this.setState({
+        counterValue: initialValue + 1
+    });
+}
+decrementCounter(){
+    let initialValue = this.state.counterValue;
+    this.setState({
+        counterValue: initialValue - 1
+    });
+}
+render(){
+    return (
+        <div className = "regularDiv">
+            <h1>Counter: {this.state.counterValue}</h1>
+            <button onClick={this.incrementCounter.bind(this)}>increment</button>
```

Comment:
I would advise to bind the methods in the constructor or use other methods of handling this.
Please review this page: [https://reactjs.org/docs/handling-events.html](https://reactjs.org/docs/handling-events.html)

Reply:
thanks changed